dot/network/block_announce.go

```diff
@@ -203,14 +203,13 @@ func (s *Service) validateBlockAnnounceHandshake(from peer.ID, hs Handshake) err
 // if some more blocks are required to sync the announced block, the node will open a sync stream
 // with its peer and send a BlockRequest message
 func (s *Service) handleBlockAnnounceMessage(from peer.ID, msg NotificationsMessage) (propagate bool, err error) {
+	propagate = false
+
 	bam, ok := msg.(*BlockAnnounceMessage)
 	if !ok {
-		return false, errors.New("invalid message")
-	}
-
-	if err = s.syncer.HandleBlockAnnounce(from, bam); err != nil {
-		return false, err
+		return propagate, errors.New("invalid message")
```

Comment:
Can this error ever happen? Shouldn't we panic instead here? :thinking: 
Especially since the function is named `handleBlockAnnounceMessage`

Reply:
> Can this error ever happen?

I would say no. This function (`handleBlockAnnounceMessage`) is called after we call the `decodeBlockAnnounceMessage` which returns a `*BlockAnnounceMessage`, so in a case where we receive a wrong block announce message the function `decodeBlockAnnounceMessage` will fail and return an error and then the function `handleBlockAnnounceMessage` will never be called. I can remove this guard check